apis/fund_api_models.py

```diff
@@ -0,0 +1,66 @@
+"""Contains the set up for our API and a basic DAO
+for prototyping.
+"""
+from flask_restx import fields
+from flask_restx import Namespace
+
+api = Namespace("funds", description="Operations related to fund infomation")
+
+eligibility_criteria_sub_model = api.model(
+    "Eligibility",
+    {
+        "maximium_project_cost": fields.Integer(
+            description="The maxmium amount a project can ask for"
+        )
+    },
+)
+
+round_sub_model = api.model(
+    "Fund Round",
+    {
+        "round_identifier": fields.Integer(
+            required=True,
+            description="An unique integer which identifies the round.",
+        ),
+        "opens": fields.DateTime(
+            required=True, description="The round opening date"
+        ),
+        "deadline": fields.DateTime(description="The round closing date"),
+    },
+)
+
+full_fund_model = api.model(
+    "Fund Full Data",
+    {
+        "name": fields.String(
+            required=True, description="The name of the fund"
+        ),
+        "fund_identifier": fields.String(
+            required=True, description="The unique id for this fund"
+        ),
+        "eligibility_criteria": fields.Nested(
+            eligibility_criteria_sub_model,
+            desciption="The eligiblity criteria of the fund",
+        ),
+        "rounds": fields.List(
+            fields.Nested(round_sub_model),
+            required=True,
+            description=(
+                "A list of the funding rounds, along with data regarding their"
+                " opening and closing dates."
+            ),
+        ),
+    },
+)
+
+identify_fund_view = api.model(
+    "Fund Summary",
+    {
+        "name": fields.String(
+            required=True, description="The name of the fund"
+        ),
+        "fund_identifier": fields.String(
+            required=True, description="The unique id for this fund"
+        ),
+    },
+)
```

Comment:
This is more of a comment/question about flask_restx. The requirement to have model-like api objects which are not the same as (nor as straightforward as) data classes makes the code a bit hard to digest. It also may create potential for confusion and errors to creep in as they drift further from the underlying data models? Is there a simpler way, perhaps worth a discussion about pydantic or FastAPI?

Reply:
I agree about having a discussion. 

@vialde @Adam-W1  Any strong opinions on flask_restx vs FastAPI? It would be better for dev experience to have a single library in use - the whole point of flask is to add little bits to it as needed - but I do agree that the data validation is quite unwieldy. 

Although given that the data in each micro service is quite flat and probably won't become much more complicated then this is probably not actually a major issue? 

Just some thoughts.

Reply:
some good points across the comments guys. I don't have much experience in either framework so i couldn't comment on that comparison specifically. I think the flast_restx could be more intuitive and if there is a better option out there then it would make sense to shift now whilst we are early days, however, the model like api objects may become more intuitive when sat above an actual database - it may then provide a more consistent picture.

Reply:
The model objects are very close to json data schema - as time goes on it might actually become more intuitive as Adam has said. 

At least if we stick to flask_restx then we can easily create a good mental model of how flask works in various ways...thus our development velocity would increase. 

Including more and more libraries might lead to more context switching...[not good](https://www.researchgate.net/publication/317989659_Impact_of_task_switching_and_work_interruptions_on_software_development_processes)!